### PR TITLE
[Kotlin migration ] AnalyticsTest 

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Kotlin migration of AnalyticsTest.java

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison — CHANGES_REQUESTED** [review]
Tests failing. Please rebase this onto `main` as we had a relevant PR just go in.

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 1]
```diff
@@ -1,158 +1,132 @@
...
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+class AnalyticsTest {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:36]: `cleanup: when -> whenever`

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 2]
```diff
@@ -1,158 +1,132 @@
...
-                .thenReturn(mMockSharedPreferencesEditor);
-
+    fun setUp() {
+        UsageAnalytics.resetForTests()
+        MockitoAnnotations.openMocks(this)
+        Mockito.`when`(mMockResources!!.getBoolean(R.bool.ga_anonymizeIp))
+            .thenReturn(true)
+        Mockito.`when`(mMockResources.getInteger(R.integer.ga_sampleFrequency))
+            .thenReturn(10)
+        Mockito.`when`(mMockContext!!.resources)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:56]: fix (in this commit): import `Mockito`

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 3]
```diff
@@ -1,158 +1,132 @@
...
-            ExceptionHit spyHit = spy(new ExceptionHit());
-            doReturn(spyHit).when(analytics).exception();
-
-            try {
-                UsageAnalytics.sendAnalyticsException(grandparentException, false);
-            } catch (Exception e) {
-                // do nothing - this is expected because UsageAnalytics isn't fully initialized
+    @Test // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
+    fun testSendException() {
+        Mockito.mockStatic(PreferenceManager::class.java).use { _ ->
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:88]: I'm surprised this works in Kotlin, can you check the tests in a debugger

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt (hunk 2 above)
Comment (Prince-kushwaha) [reply AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:56]: i not understand what i have to do?

Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:56]: Import `Mockito`, so the line starts with 
```
`when`(mMockContext!!.resources)
```

**Comment by david-allison** [comment]
WARN: conflicts with https://github.com/ankidroid/Anki-Android/pull/10864

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt (hunk 2 above)
Comment (Prince-kushwaha) [reply AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:56]: i have cleanup: when -> whenever

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 4]
```diff
@@ -1,158 +1,133 @@
...
-        when(mMockSharedPreferencesEditor.putBoolean(UsageAnalytics.ANALYTICS_OPTIN_KEY, true))
-                .thenReturn(mMockSharedPreferencesEditor);
-
-        when(mMockSharedPreferences.edit())
-                .thenReturn(mMockSharedPreferencesEditor);
-
+    fun setUp() {
+        UsageAnalytics.resetForTests()
+        MockitoAnnotations.openMocks(this)
+        whenever((mMockResources!!.getBoolean(R.bool.ga_anonymizeIp)))
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:53]: Please use `when`. The second commit should be the minimum effort required to get CI passing, keeping as true to CI as possible 

With the exception of imports, which want to be as true to the Java as possible - the Kotlin migration messes up imports, adding or removing them in a lot of cases

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 5]
```diff
@@ -1,158 +1,132 @@
...
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+@file:Suppress("DEPRECATION")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:16]: Move this back to `testSendException`

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 6]
```diff
@@ -1,158 +1,132 @@
...
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers
+import org.mockito.Mock
+import org.mockito.Mockito.*
+import org.mockito.MockitoAnnotations
+
+class AnalyticsTest {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:36]: Add a `KotlinCleanup` for `when` usage to be converted to `whenever`

File: AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt [hunk 7]
```diff
@@ -1,158 +1,132 @@
...
     @Mock
-    private Context mMockContext;
+    private val mMockContext: Context? = null
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnalyticsTest.kt:38]: KotlinCleanup to see if these can be made `lateinit`

**Review by david-allison — COMMENTED** [review]
Looks great! Think this is going to be it

**Review by david-allison — APPROVED** [review]

**Comment by david-allison** [comment]
Great! Thanks

**Review by mikehardy — APPROVED** [review]
Nice, thank you!

**Pull request merged**